### PR TITLE
feat: add a signMessage function

### PR DESCRIPTION
**Pull request by gtsonevv**
## Pre-flight checklist

- [ ] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [ ] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [ ] **If this is a code change**: I have written unit tests.
- [ ] **If this changes code in a published package**: I have run `pnpm changeset` to create a `changeset` JSON document appropriate for this change.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

### CODE REVIEW

File: packages/accounts/src/account.ts [hunk 1]
```diff
@@ -711,4 +727,29 @@ export class Account {
...
+     * @param message The message that wants to be transmitted.
+     * @param recipient The recipient to whom the message is destined (e.g. "alice.near" or "myapp.com").
+     * @param nonce A nonce that uniquely identifies this instance of the message, denoted as a 32 bytes array (a fixed `Buffer` in JS/TS).
+     * @param callbackUrl Optional, applicable to browser wallets (e.g. MyNearWallet). The URL to call after the signing process. Defaults to `window.location.href`.
+     */
+    async signMessage({ message, recipient, nonce, callbackUrl }: SignMessageParams): Promise<SignedMessage> {
+        // Check the nonce is a 32bytes array
+        if (nonce.length != 32) { throw Error('Expected nonce to be a 32 bytes buffer'); }
+        
+        const tag = 2147484061;
```
Comment (vikinatora) [inline packages/accounts/src/account.ts]: This looks to me like a magic number. Please give it some context or extract it to a constant somewhere

File: packages/accounts/test/test-utils.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
-const { KeyPair } = require('@near-js/crypto');
+const { KeyPair, PublicKey } = require('@near-js/crypto');
 const { InMemoryKeyStore } = require('@near-js/keystores');
 const BN = require('bn.js');
 const fs = require('fs').promises;
+const { serialize } = require('borsh');
+const { sha256 } = require('js-sha256');
 
 const { Account, AccountMultisig, Contract, Connection, LocalAccountCreator } = require('../lib');
+const { SCHEMA } = require( '@near-js/transactions' );
```
Comment (vikinatora) [inline packages/accounts/test/test-utils.js]: This import should be moved to the top where the other `@near-js` imports are

File: packages/accounts/test/test-utils.js [hunk 3]
```diff
@@ -112,6 +115,20 @@ function waitFor(fn) {
     return _waitFor();
 }
 
+function verifySignature({ message, nonce, recipient, callbackUrl, publicKey, signature }) {
+    const tag = 2147484061;
```
Comment (vikinatora) [inline packages/accounts/test/test-utils.js]: Again the magic number

File: packages/accounts/test/test-utils.js [hunk 4]
```diff
@@ -112,6 +115,20 @@ function waitFor(fn) {
...
+    const borshPayload = serialize(SCHEMA.SignMessagePayload, {
+        tag,
+        message,
+        nonce,
+        recipient,
+        callbackUrl
+    });
+    const toSign = Uint8Array.from(sha256.array(borshPayload));
+    const myPK = PublicKey.from(publicKey);
+    return myPK.verify(toSign, Buffer.from(signature, 'base64'));
```
Comment (vikinatora) [inline packages/accounts/test/test-utils.js]:
```suggestion
    const pk = PublicKey.from(publicKey);
    return pk.verify(toSign, Buffer.from(signature, 'base64'));
```

**Review by vikinatora — APPROVED** [review]

File: packages/accounts/src/account.ts (hunk 1 above)
Comment (gtsonevv) [reply packages/accounts/src/account.ts]: will extract it to a constant

**Pull request merged**